packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -122,7 +122,23 @@ abstract class FlutterCommand extends Command<Null> {
     _usesPubOption = true;
   }
 
-  void addBuildModeFlags({ bool defaultToRelease: true }) {
+  void usesBuildNumberOption() {
+    argParser.addOption('buildNumber',
```

Comment:
Flutter tool arguments are of the form _hyphen-case_, so this would be `'build-number'`